Add more context flags and getters to `MeasuredTemplateDocumentPF2e`

In3luki opened the pull request.

stwlam commented:
Can `MeasuredTemplateDocument` be constructed with a flag such that it knows what kind of pathfinder area it is?

In3luki commented:
I have removed the `drawPreview` arguments and moved the new context to `MeasuredTemplateDocumentPF2e#flags`.
I've also created some new getters on the document and the template object for easier access of the flags.

stwlam commented on src/module/canvas/measured-template.ts:
```diff
@@ -48,18 +66,17 @@ class MeasuredTemplatePF2e<
         if (canvas.grid.type === CONST.GRID_TYPES.SQUARE) {
-            const { item } = this.document;
-            if (item?.isOfType("spell") && item.system.area?.type === "burst") {
+            if (this.spellArea?.type === "burst") {
```
```suggestion
            if (this.effectArea?.type === "burst") {
```

stwlam commented on src/module/item/spell/document.ts:
```diff
@@ -505,6 +505,7 @@ class SpellPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Ite
                         traits: deepClone(this.system.traits.value),
                         ...this.getOriginData(),
                     },
+                    spellArea: this.system.area,
```
```suggestion
                    effectArea: this.system.area,
```

stwlam commented on src/module/scene/measured-template-document.ts:
```diff
@@ -48,14 +60,17 @@ export class MeasuredTemplateDocumentPF2e<
         pf2e?: {
             messageId?: string;
             origin?: ItemOriginFlag;
+            spellArea?: Maybe<SpellArea>;
```
```suggestion
            effectArea?: Maybe<SpellArea>;
```

stwlam commented on src/module/scene/measured-template-document.ts:
```diff
@@ -31,6 +39,10 @@ export class MeasuredTemplateDocumentPF2e<
         return game.messages.get(this.flags.pf2e?.messageId ?? "") ?? null;
     }
 
+    get spellArea(): SpellArea | null {
+        return this.flags.pf2e?.spellArea ?? null;
+    }
```
```suggestion
    get effectArea(): SpellArea | null {
        return this.flags.pf2e?.effectArea ?? null;
    }
```

stwlam commented on src/module/canvas/measured-template.ts:
```diff
@@ -19,6 +21,22 @@ class MeasuredTemplatePF2e<
+        return this.document.message;
+    }
+
+    get spellArea(): SpellArea | null {
+        return this.document.spellArea;
+    }
```
```suggestion
    get effectArea(): SpellArea | null {
        return this.document.effectArea;
    }
```

In3luki commented:
Updated
